### PR TITLE
Add a new state to termniate a server without making an archive

### PR DESCRIPTION
**Pull request by g1franc**
##### SUMMARY
Add a new state to termniate a server without making an archive.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
- scaleway_compute

##### ANSIBLE VERSION

```
ansible 2.7.0.a1.post0
  config file = None
  configured module search path = [u'/home/francogu/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.14 (default, Oct 31 2017, 21:12:13) [GCC 6.4.0]
```

### CODE REVIEW

**Comment by g1franc** [comment]
@sieben

**Comment by g1franc** [comment]
@sieben Could you validate it so it can progress in the merge workflow ?

File: lib/ansible/modules/cloud/scaleway/scaleway_compute.py [hunk 1]
```diff
@@ -59,6 +61,7 @@
       - running
       - restarted
       - stopped
+      - terminate
```
Comment (remyleone) [inline lib/ansible/modules/cloud/scaleway/scaleway_compute.py]: terminated. Also it is not added in SCALEWAY_SERVER_STATES neither in SCALEWAY_TRANSITIONS_STATES (terminating)?

File: lib/ansible/modules/cloud/scaleway/scaleway_compute.py [hunk 2]
```diff
@@ -502,7 +536,8 @@ def restart_strategy(compute_api, wished_server):
     "restarted": restart_strategy,
     "stopped": stop_strategy,
     "running": running_strategy,
-    "absent": absent_strategy
+    "absent": absent_strategy,
+    "terminate": terminate_strategy
```
Comment (remyleone) [inline lib/ansible/modules/cloud/scaleway/scaleway_compute.py]: terminated

**Review by remyleone — CHANGES_REQUESTED** [review]
Apart from changing from terminate to terminated for consistency, this PR is good to go. LGTM

**Comment by remyleone** [comment]
Tested using this:

```
---

- name: foobar
  hosts: localhost
  vars:
    scaleway_image_id: 89ee4018-f8c3-4dc4-a6b5-bca14f985ebe
    scaleway_region: ams1
    scaleway_commerial_type: START1-S
    scaleway_organization: "{{ lookup('env', 'SCW_ORG') }}"

  tasks:
  - name: Run a compute instance
    scaleway_compute:
      name: foobar
      state: running
      image: '{{ scaleway_image_id }}'
      organization: '{{ scaleway_organization }}'
      region: '{{ scaleway_region }}'
      commercial_type: '{{ scaleway_commerial_type }}'
      wait: true

  - name: terminate the instance
    scaleway_compute:
      name: foobar
      state: terminated
      image: '{{ scaleway_image_id }}'
      organization: '{{ scaleway_organization }}'
      region: '{{ scaleway_region }}'
      commercial_type: '{{ scaleway_commerial_type }}'
      wait: true
```

File: lib/ansible/modules/cloud/scaleway/scaleway_compute.py [hunk 3]
```diff
@@ -565,7 +600,7 @@ def server_change_attributes(compute_api, target_server, wished_server):
 def core(module):
     region = module.params["region"]
     wished_server = {
-        "state": module.params["state"],
+        "state": "absent" if module.params["state"] == "terminate" else module.params["state"],
```
Comment (mscherer) [inline lib/ansible/modules/cloud/scaleway/scaleway_compute.py]: I guess "terminated" here too.

**Comment by remyleone** [comment]
@mscherer Is it good to go?

File: lib/ansible/modules/cloud/scaleway/scaleway_compute.py [hunk 4]
```diff
@@ -565,7 +600,7 @@ def server_change_attributes(compute_api, target_server, wished_server):
 def core(module):
     region = module.params["region"]
     wished_server = {
-        "state": module.params["state"],
+        "state": "absent" if module.params["state"] == "terminated" else module.params["state"],
```
Comment (remyleone) [inline lib/ansible/modules/cloud/scaleway/scaleway_compute.py:603]: @g1franc I'm not sure about this change. Why is it here?

Comment (g1franc) [reply lib/ansible/modules/cloud/scaleway/scaleway_compute.py:603]: It uses the value received from the state argument. As it was changed to "terminated", this condition has to be updated too.

File: lib/ansible/modules/cloud/scaleway/scaleway_compute.py [hunk 5]
```diff
@@ -358,6 +365,33 @@ def absent_strategy(compute_api, wished_server):
...
+
+    if compute_api.module.check_mode:
+        return changed, {"status": "Server %s would be made terminated." % target_server["id"]}
+
+    # Here we should check that the server is running and not locked.
+    # Terminate a server WITHOUT snapshot before
+    response = terminate_server(compute_api=compute_api, server=target_server,)
+
+    if not response.ok:
+        err_msg = 'Error while removing server [{0}: {1}]'.format(response.status_code, response.json)
```
Comment (mscherer) [inline lib/ansible/modules/cloud/scaleway/scaleway_compute.py:389]: So, are we sure the response is going to be json ? Like, if the server fail to answer ?

**Review by mscherer — COMMENTED** [review]
Some notes, we had the issue on other scaleway modules I think

**Pull request closed without merging**